File: dolmisani.games.graphrace/src/dolmisani/games/graphrace/Step.java
package dolmisani.games.graphrace;

public class Step {
	
	private int deltaX;
	private int deltaY;

	public Step(int deltaX, int deltaY) {

		this.deltaX = deltaX;
		this.deltaY = deltaY;
	}

	public double length() {
		return Math.sqrt(deltaX*deltaX + deltaY*deltaY);
	}


	public int getDeltaX() {
		
		return deltaX;
	}

	public int getDeltaY() {
		
		return deltaY;
	}

	public void setDeltaX(int deltaX) {
		
		this.deltaX = deltaX;
	}

	public void setDeltaY(int deltaY) {
		
		this.deltaY = deltaY;
	}

	@Override
	public String toString() {
		
		return String.format("[%d, %d]", deltaX, deltaY);
	}
}

File: pwa/js/step.js
class Step {
    constructor(deltaX, deltaY) {
        this.deltaX = deltaX;
        this.deltaY = deltaY;
    }

    length() {
        return Math.sqrt(this.deltaX * this.deltaX + this.deltaY * this.deltaY);
    }

    getDeltaX() {
        return this.deltaX;
    }

    getDeltaY() {
        return this.deltaY;
    }

    setDeltaX(deltaX) {
        this.deltaX = deltaX;
    }

    setDeltaY(deltaY) {
        this.deltaY = deltaY;
    }

    toString() {
        return `(${this.deltaX}, ${this.deltaY})`;
    }
}
